src/Temporalio/Bridge/OptionsExtensions.cs

```diff
@@ -450,16 +450,66 @@ public static Interop.WorkerOptions ToInteropOptions(
             {
                 throw new ArgumentException("Task queue must be provided in worker options");
             }
-            var buildId = options.BuildId;
+#pragma warning disable 0618
+            var buildId = options.DeploymentOptions?.Version?.BuildId ?? options.BuildId;
+#pragma warning restore 0618
+            var buildIdAutoDetermined = false;
             if (buildId == null)
             {
-                if (options.UseWorkerVersioning)
-                {
-                    throw new ArgumentException("BuildId must be explicitly set when UseWorkerVersioning is true");
-                }
                 var entryAssembly = Assembly.GetEntryAssembly() ??
                     throw new ArgumentException("Unable to get assembly manifest ID for build ID");
                 buildId = entryAssembly.ManifestModule.ModuleVersionId.ToString();
+                buildIdAutoDetermined = true;
+            }
+            Interop.WorkerVersioningStrategy versioningStrategy;
+            if (options.DeploymentOptions != null)
+            {
+                if (options.DeploymentOptions.Version == null)
```

Comment:
Should we also check that `options.UseWorkerVersioning` is `false`?

Reply:
It's checked now